hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CustomGasCharging.java

```diff
@@ -129,6 +129,11 @@ public GasCharges chargeForGas(
         if (transaction.isEthereumTransaction()) {
             requireNonNull(relayer);
             final var allowanceUsed = chargeWithRelayer(sender, relayer, context, worldUpdater, transaction);
+
+            // Increment nonce right before the gas is charged
```

Comment:
This says to increment "right before"; but it's clearly being incremented right _after_ charging. 😁 

Is the comment wrong, or the order of operations wrong?